pkg/clusteragent/admission/controllers/webhook/controller_base_test.go

```diff
@@ -62,3 +72,95 @@ func TestNewController(t *testing.T) {
 
 	assert.IsType(t, &ControllerV1beta1{}, controller)
 }
+
+func TestAutoInstrumentation(t *testing.T) {
+	tests := []struct {
+		name        string
+		pod         *corev1.Pod
+		config      string
+		expectPatch bool
+	}{
+		{
+			name:   "disabled webhooks should not patch",
+			config: "testdata/all_disabled.yaml",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+					Labels: map[string]string{
+						"admission.datadoghq.com/enabled": "true",
+					},
+					Namespace: "foo",
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: "foo",
+						},
+					},
+				},
+			},
+			expectPatch: false,
+		},
+		{
+			name:   "disabled webhooks should not patch, even with targets",
+			config: "testdata/all_disabled_targets.yaml",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+					Labels: map[string]string{
+						"admission.datadoghq.com/enabled": "true",
+					},
+					Namespace: "foo",
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: "foo",
+						},
+					},
+				},
+			},
+			expectPatch: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create a mock config.
+			mockConfig := configmock.NewFromFile(t, tt.config)
+
+			// Create a mock meta.
+			wmeta := fxutil.Test[workloadmetamock.Mock](t, fx.Options(
+				fx.Supply(coreconfig.Params{}),
+				fx.Provide(func() log.Component { return logmock.New(t) }),
+				coreconfig.MockModule(),
+				workloadmetafxmock.MockModule(workloadmeta.NewParams()),
+			))
+
+			// Create APM webhook.
+			apm, err := generateAutoInstrumentationWebhook(wmeta, mockConfig)
+			assert.NoError(t, err)
+
+			// Create request.
+			podJSON, err := json.Marshal(tt.pod)
+			assert.NoError(t, err)
+			t.Log(string(podJSON))
+			request := &admission.Request{
+				Object:    podJSON,
+				Namespace: tt.pod.Namespace,
+			}
+
+			// Send request.
+			f := apm.WebhookFunc()
+			response := f(request)
+
+			// Check if the patch is expected.
+			emptyPatch := []byte("null")
```

Comment:
Should this just be `"null"` since we're casting the results to strings in the assertion?